### PR TITLE
add fakes for events package, add startEventWatcher to event interface

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: split changes of events from #78447

**Special notes for your reviewer**:

/assign @wojtek-t 
/priority important-soon
/sig scalability

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/events/fake.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+// FakeRecorder is used as a fake during tests. It is thread safe. It is usable
+// when created manually and not by NewFakeRecorder, however all events may be
+// thrown away in this case.
+type FakeRecorder struct {
+	Events chan string
+}
+
+//Eventf emits an event
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/tools/events/fake.go]: nit: space after "//"
// Eventf ...

File: staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/fake/fake_event_expansion.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2014 The Kubernetes Authors.
+Copyright 2019 The Kubernetes Authors.
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/fake/fake_event_expansion.go]: Why?

**Review by wojtek-t — COMMENTED** [review]
Just small nits.

File: staging/src/k8s.io/client-go/tools/events/fake.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+func (f *FakeRecorder) Eventf(regarding runtime.Object, related runtime.Object, eventtype, reason, action, note string, args ...interface{}) {
+	if f.Events != nil {
+		f.Events <- fmt.Sprintf(eventtype+" "+reason+" "+note, args...)
+	}
+}
+
+// NewFakeRecorder creates new fake event recorder with event channel with
+// buffer of given size.
+func NewFakeRecorder(bufferSize int) *FakeRecorder {
```
Comment (yue9944882) [inline staging/src/k8s.io/client-go/tools/events/fake.go]: whats the point of passing buffer size here?

File: staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/fake/fake_event_expansion.go (hunk 2 above)
Comment (yastij) [reply staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/fake/fake_event_expansion.go]: it's something written in 2019, that's why

File: staging/src/k8s.io/client-go/tools/events/fake.go (hunk 3 above)
Comment (yastij) [reply staging/src/k8s.io/client-go/tools/events/fake.go]: mostly to handle back-pressure in a flexible way for tests

File: staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/fake/fake_event_expansion.go (hunk 2 above)
Comment (yliaog) [inline staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/fake/fake_event_expansion.go]: it should stay as 2014

**Pull request merged**